test/unit/int_algolia/scripts/algolia/model/algoliaLocalizedProduct.test.js

```diff
@@ -504,3 +504,59 @@ describe('algoliaLocalizedProduct overriding custom attributes', function () {
         expect(new AlgoliaLocalizedProductModel({ product: product, locale: 'fr', attributeList: ['algoliaTest', 'custom.displaySize'], baseModel: baseModel})).toEqual(expectedProductModel);
     });
 });
+
+describe('IndexOutofStock logic tests', function () {
+    var ProductMockObj;
+
+    beforeEach(() => {
+        jest.resetModules();
+        ProductMockObj = require('../../../../../mocks/dw/catalog/Variant');
+    });
+
+    test('should skip out-of-stock variants if IndexOutofStock = false', function () {
+        jest.doMock('*/cartridge/scripts/algolia/lib/algoliaData', () => {
+            return {
+                getSetOfArray: function () {
+                    return []; 
+                },
+                getPreference: function (id) {
+                    if (id === 'InStockThreshold') return 1;
+                    if (id === 'IndexOutofStock') return false;
+                    return null;
+                }
+            };
+        }, { virtual: true });
+
+
+        const product = new ProductMockObj({
+            inventory: { atsValue: 0 }
+        });
+
+        const localizedProduct = new AlgoliaLocalizedProduct({ product, attributeList: ['objectID'] });
+
+        expect(localizedProduct.objectID).toBeNull();
```

Comment:
? The test doesn't check anything regarding skipping out-of-stock products.

Here you are generating an AlgoliaProduct from a SFCC product, with the `objectID` attribute. But `objectID` doesn't exist in SFCC products, so it will necessarily be null :shrug: 

Please take the time to run step by step it in debug to understand what is happening.